### ISSUE TITLE
Random `NotFittedError`s with `rotbaum.TreeEstimator`

### ISSUE THREAD

**Issue by caseytomlin** [awslabs/gluon-ts#2483]
## Description
Executing the code below _sometimes_ succeeds, _sometimes_ fails, with a `NotFittedError`. I am running it in a notebook cell, and it does not seem to change whether or not I restart the kernel.

## To Reproduce
As the title suggests, no guarantee....

Mostly lifted from [this passing test](https://github.com/awslabs/gluonts/blob/dev/test/ext/rotbaum/test_rotbaum_smoke.py):
```python
from gluonts.testutil.dummy_datasets import make_dummy_datasets_with_features
from gluonts.model.rotbaum import TreeEstimator

datasets = make_dummy_datasets_with_features(
    cardinality=[3, 10, 42], num_feat_dynamic_real=3
)

dataset_train, dataset_test = datasets
hps = {
    "freq": "D",
    "prediction_length": 3,
    "context_length": 3,
}
estimator = TreeEstimator.from_inputs(dataset_train, **hps)
predictor = estimator.train(dataset_train)

forecasts = list(predictor.predict(dataset_test))
forecasts[0].median
```

## Error message or code output

```
---------------------------------------------------------------------------
NotFittedError                            Traceback (most recent call last)
Cell In[5], line 27
     19 # forecast_it, ts_it = make_evaluation_predictions(
     20 #     dataset=dataset_test,
     21 #     predictor=predictor
     22 # )
     23 # fcs = list(forecast_it)
     24 # fcs[0].quantile_ts(q=0.5)
     26 forecasts = list(predictor.predict(dataset_test))
---> 27 forecasts[0].median

File ~/.venv/lib/python3.9/site-packages/gluonts/model/forecast.py:149, in Forecast.median(self)
    147 @property
    148 def median(self) -> np.ndarray:
--> 149     return self.quantile(0.5)

File ~/.venv/lib/python3.9/site-packages/gluonts/model/rotbaum/_predictor.py:68, in RotbaumForecast.quantile(self, q)
     61 """
     62 Returns np.array, where the i^th entry is the estimate of the q
     63 quantile of the conditional distribution of the value of the i^th
     64 step in the forecast horizon.
     65 """
     66 assert 0 <= q <= 1
     67 return np.array(
---> 68     list(
     69         chain.from_iterable(
     70             model.predict(self.featurized_data, q)
     71             for model in self.models
     72         )
     73     )
     74 )

File ~/.venv/lib/python3.9/site-packages/gluonts/model/rotbaum/_predictor.py:70, in <genexpr>(.0)
     61 """
     62 Returns np.array, where the i^th entry is the estimate of the q
     63 quantile of the conditional distribution of the value of the i^th
     64 step in the forecast horizon.
     65 """
     66 assert 0 <= q <= 1
     67 return np.array(
     68     list(
     69         chain.from_iterable(
---> 70             model.predict(self.featurized_data, q)
     71             for model in self.models
     72         )
     73     )
     74 )

File ~/.venv/lib/python3.9/site-packages/gluonts/model/rotbaum/_model.py:373, in QRX.predict(self, x_test, quantile)
    371 predicted_values = []
    372 for pt in x_test:
--> 373     pred = self.model.predict(np.array([pt]))[
    374         0
    375     ]  # xgboost doesn't like lists
    376     closest_pred = self.get_closest_pt(
    377         self.sorted_train_preds, pred
    378     )
    379     predicted_values.append(
    380         self._get_and_cache_quantile_computation(
    381             closest_pred, quantile
    382         )
    383     )

File ~/.venv/lib/python3.9/site-packages/xgboost/sklearn.py:1135, in XGBModel.predict(self, X, output_margin, ntree_limit, validate_features, base_margin, iteration_range)
   1100 """Predict with `X`.  If the model is trained with early stopping, then `best_iteration`
   1101 is used automatically.  For tree models, when data is on GPU, like cupy array or
   1102 cuDF dataframe and `predictor` is not specified, the prediction is run on GPU
   (...)
   1131 
   1132 """
   1133 with config_context(verbosity=self.verbosity):
   1134     iteration_range = _convert_ntree_limit(
-> 1135         self.get_booster(), ntree_limit, iteration_range
   1136     )
   1137     iteration_range = self._get_iteration_range(iteration_range)
   1138     if self._can_use_inplace_predict():

File ~/.venv/lib/python3.9/site-packages/xgboost/sklearn.py:649, in XGBModel.get_booster(self)
    646 if not self.__sklearn_is_fitted__():
    647     from sklearn.exceptions import NotFittedError
--> 649     raise NotFittedError("need to call fit or load_model beforehand")
    650 return self._Booster

NotFittedError: need to call fit or load_model beforehand
```

## Environment
- Operating system: Linux
- Python version: 3.9.12
- GluonTS version: 0.9.9
- MXNet version: 1.9.1
- xgboost version: 1.7.1

**Comment by lostella** [comment]
@zoolhasson have you ever encountered this?

**Comment by lostella** [comment]
@caseytomlin #2537 fixes the issue, and will be included in the upcoming 0.12 release! You can in the mean time test it out by installing the `dev` branch if you want.

**Comment by caseytomlin** [comment]
thanks @zoolhasson and @lostella !
